content/specs/auth.md

```diff
@@ -0,0 +1,390 @@
+
+After the auth flow continues and an initial token request succeeds, the client will parse the account identifier from the `sub` field in the token response. At this point, the client can still not trust that it has actually authenticated the indicated account. It is critical for the client to resolve the identity (DID document), extract the declared PDS host, and confirm that the PDS (Resource Server) is resolves to the the Authorization Server bound to the session, by fetching the Resource Server metadata.
+
+To re-iterate, it is critical for all clients - including those only interested in atproto Identity Authentication - to go through to the entire Authorization flow, and to verify that the account identifier (DID) in the `sub` field of the token response is consistent with the Authorization Server hostname/origin (`issuer`).
+
+### Authorization Scopes
+
+OAuth scopes allow more granular control over the resources and actions a client is granted access to.
+
+The special `atproto` scope is required for all atproto OAuth sessions. The semantics are somewhat similar to the `openid` scope: inclusion of it confirms that the client is using the atproto profile of OAuth and will comply with all the requirements laid out in this specification. No access to any atproto-specific PDS resources will be granted without this scope included.
+
+Authorization Servers may support other profiles of OAuth if client does not include the `atproto` scope. For example, an Authorization Server might function as both an atproto PDS/entryway, and support other protocols/standards at the same time.
+
+Use of the atproto OAuth profile, as indicated by the `atproto` scope, means that the Authorization Server will return the atproto account DID as an account identifier in the `sub` field of token requests. A client may include only the `atproto` scope if they only need account authentication, for example a “Login with atproto” use case. Unlike OpenID, profile metadata in atproto is generally public, so an additional authorization scope for fetching profile metadata is not needed.
+
+In OAuth 2.0, Authorization Servers may always return the granted scopes in auth request responses and token responses, but are only required to do so if the scope is different from what the client requested. In the atproto OAuth profile, servers must always return the granted scopes.
+
+TODO: bluesky PDS/entryway doesn't always return scopes yet
+
+The intention is to support a flexible scopes based on Lexicon namespaces (NSIDs) so that clients can be given access only to the specific content and API endpoints they need access to. Until the design of that scope system is ready, the atproto profile of OAuth defines two transitional scopes, which align with the permissions granted under the original “session token” auth system:
```

Comment:
```suggestion
The intention is to support flexible scopes based on Lexicon namespaces (NSIDs) so that clients can be given access only to the specific content and API endpoints they need access to. Until the design of that scope system is ready, the atproto profile of OAuth defines two transitional scopes which align with the permissions granted under the original “session token” auth system:
```